src/nodes.jl

```diff
@@ -616,7 +616,8 @@ Print a simplified string representation of the `CollectNode` with its nodes Vec
 result DeferredFuture RemoteChannel parameters, and its label.
 """
 function Base.show(io::IO, node::CollectNode)
-    print(io, "$(typeof(node).name.name)($(node.nodes),$(node.result),\"$(node.label)\")")
+    nodes = "DispatchNode[$(join(node.nodes, ","))]"
```

Comment:
adding the io to a join inside a string seems to not do what we want:
`"NodeSet(DispatchNode[nothing])"` as opposed to  `"NodeSet(DispatchNode[])"`